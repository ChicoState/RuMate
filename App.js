import React from 'react';
import firebase from 'firebase';
import { StyleSheet } from 'react-native';
import { useScreens } from 'react-native-screens';
import { createAppContainer, createSwitchNavigator } from 'react-navigation';
import { createBottomTabNavigator } from 'react-navigation-tabs';
import { createStackNavigator } from 'react-navigation-stack';
import Icon from 'react-native-vector-icons/FontAwesome';
/* our imports */
import HomeScreen from './src/screens/HomeScreen';
import TasksScreen from './src/screens/TasksScreen';
import AuthScreen from './src/screens/AuthScreen';
import CreateTaskScreen from './src/screens/CreateTaskScreen';
import AddRoommateScreen from './src/screens/AddRoomateScreen';
import CalendarScreen from './src/screens/CalendarScreen';
import BillsScreen from './src/screens/BillsScreen';
import CreateBillScreen from './src/screens/CreateBillScreen';
import CreateRoommateGroupScreen from './src/screens/CreateRoommateGroupScreen';
import MessageScreen from './src/screens/MessagesScreen';
import CreateConversationScreen from './src/screens/CreateConversationScreen';
import ConversationScreen from './src/screens/ConversationScreen';
import InvitationsScreen from './src/screens/InvitationsScreen';
import AccountScreen from './src/screens/AccountScreen';
import PriorityScreen from './src/screens/PriorityScreen';
import ChangeDetailsScreen from './src/screens/ChangeDetailsScreen';

useScreens();
// ################################ //
// Remove to debug errors/warnings  //
   console.disableYellowBox = true; //
// ################################ //
const firebaseConfig = {
  apiKey: "AIzaSyA_ZWMR-MFG_ZHNK4_WuEeHoLP9vzsY_Vk",
  authDomain: "rumate-faaeb.firebaseapp.com",
  databaseURL: "https://rumate-faaeb.firebaseio.com",
  projectId: "rumate-faaeb",
  storageBucket: "",
  messagingSenderId: "484837108351",
  appId: "1:484837108351:web:53a875e9546d00aa2dad4a"
};
// Initialize Firebase
firebase.initializeApp(firebaseConfig);

const tabNavigator = createBottomTabNavigator({
  Home: {
    screen: HomeScreen,
    navigationOptions: {
      tabBarLabel: "Home",
      tabBarIcon: ({ tintColor }) => (
        <Icon name="home" size={20} />
      )
    },
  },
  Tasks: {
    screen: TasksScreen,
    navigationOptions: {
      tabBarLabel: "Tasks",
      tabBarIcon: ({ tintColor }) => (
        <Icon name="tasks" size={20} />
      )
    },
  },
  Bills: {
    screen: BillsScreen,
    navigationOptions: {
      tabBarLabel: "Bills",
      tabBarIcon: ({ tintColor }) => (
        <Icon name="dollar" size={20} />
      )
    },
  },
  Messages: {
    screen: MessageScreen,
    navigationOptions: {
      tabBarLabel: "Messages",
      tabBarIcon: ({ tintColor }) => (
        <Icon name="comments" size={20} />
      )
    },
  }
},
{
  initialRouteName: 'Home',
});

const taskNavigator = createStackNavigator({
  CreateTask: CreateTaskScreen,
}, {
  defaultNavigationOptions: {
    header: null
  }
});

const appStackNavigator = createStackNavigator({
  Priority: PriorityScreen,
  Invitations: InvitationsScreen,
  CreateRoommateGroup: CreateRoommateGroupScreen,
  AddRoommate: AddRoommateScreen,
  Home: tabNavigator,
  Tasks: taskNavigator,
  CreateBill: CreateBillScreen,
  TaskCalendar: CalendarScreen,
  Messages: MessageScreen,
  CreateConversation: CreateConversationScreen,
  Conversation: ConversationScreen,
  Account: AccountScreen,
  ChangeDetails: ChangeDetailsScreen

},{
  initialRouteName: 'Home',
  defaultNavigationOptions: {
    header: null
  }
});

const authStack = createStackNavigator({
  Login: AuthScreen,
}, {
  defaultNavigationOptions: {
    header: null
  }
});

const switchNav = createSwitchNavigator({
  App: appStackNavigator,
  Auth: authStack,
}, {
  initialRouteName: 'Auth',
  defaultNavigationOptions: {
    header: null
  }
});

const App = createAppContainer(switchNav);

const styles = StyleSheet.create({
  headerLeft: {
    fontSize: 20
  }
});

export default () => {
  return (
    <App />
  );
}
