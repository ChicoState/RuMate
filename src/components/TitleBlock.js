import React from 'react';
import { View, Text, StyleSheet } from 'react-native';

const TitleBlock = ({title, color, fontSize, paddingTop, paddingBottom, paddingLeft, paddingRight}) => {
  return (
    <View>
      <Text 
        style={[styles.header, {
          color, 
          fontSize,
          paddingTop,
          paddingBottom,
          paddingLeft,
          paddingRight
        }]}>
        {title}
      </Text>
    </View>
  );
}

const styles = StyleSheet.create({
  header: {
    alignSelf: 'center'
  },
});

export default TitleBlock;