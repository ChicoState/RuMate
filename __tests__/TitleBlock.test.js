import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App'
import TitleBlock from '../src/components/TitleBlock'

describe('<TitleBlock />', () => {
  it('has 1 child', () => {
    <App />
    const tree = renderer.create(<TitleBlock />).toJSON()
    expect(tree.children.length).toBe(1)
  });
  
  it('renders correctly', () => {
    const tree = renderer.create(<TitleBlock />).toJSON()
    expect(tree).toMatchSnapshot()
  });
});