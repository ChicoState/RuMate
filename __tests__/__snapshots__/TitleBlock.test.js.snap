// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TitleBlock /> renders correctly 1`] = `
<View>
  <Text
    style={
      Array [
        Object {
          "alignSelf": "center",
        },
        Object {
          "color": undefined,
          "fontSize": undefined,
          "paddingBottom": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
        },
      ]
    }
  />
</View>
`;
